### PR TITLE
Add VIM support!

### PR DESCRIPTION
**Pull request by vueu**
**Edit**: Just realized `prettier` got rid of some parentheses on single parameter anonymous functions, I'll bring those back in a bit for consistency!

### CODE REVIEW

File: src/components/Editor/Editor.jsx [hunk 1]
```diff
@@ -118,12 +121,17 @@ class Editor extends React.Component {
...
 
-  	options.callback(value, (errors) => updateLinting(errors));
-  }
+    options.callback(value, errors => updateLinting(errors));
+  };
+
+  /**
+   * Gets the current keyMap of the editor.
+   */
+  keyMap = () => (config()._get('editor.vim') ? 'vim' : 'sublime');
```
Comment (eveningkid) [inline src/components/Editor/Editor.jsx]: Please, access it using the following instead:
```js
config().editor.vim
```
`config()._get` is meant to be avoided as much as possible.

File: src/components/Editor/Editor.jsx [hunk 2]
```diff
@@ -152,6 +169,8 @@ class Editor extends React.Component {
     let cursor = cursorPosition;
     const selection = editor.getDoc().getSelection();
 
+    this.setState({ keyMap: this.keyMap() });
```
Comment (eveningkid) [inline src/components/Editor/Editor.jsx]: Isn't it enough to update `this.state.keyMap` on focus only? As when we use `Configurator`, we will need to focus the editor again to keep coding? If not, please let me know, I am simply asking away. :)

Comment (vueu) [reply src/components/Editor/Editor.jsx]: Yes, good catch! My bad, this was my initial placement before I found `focus`, and it didn't quite work, but looks like I forgot to remove it when I was finished.

File: src/components/Editor/Editor.jsx (hunk 1 above)
Comment (vueu) [reply src/components/Editor/Editor.jsx]: Ah! Sorry I haven't really used the `electron-store` package however I should've inferred this due to the underscore. :)

**Pull request merged**